Assignment2/Q2/src/BinaryTree.java

```diff
@@ -0,0 +1,167 @@
+/**
```

Comment:
Ugh.  Copied and pasted code is a maintenance nightmare, and is horrible to review.

Please make sure you've only got one copy of your code. If there's common infrastructure to share you can use base classes and inheritance, or utility classes or whatever. In this specific case, I'd just answer both exercises with a single source file.